devito/passes/__init__.py

```diff
@@ -39,6 +39,36 @@ def is_on_device(obj, gpu_fit):
     return all(f in gpu_fit for f in fsave)
 
 
+def is_gpu_create(obj, gpu_create):
+    """
+    True if the given object is created and not copied in the device memory,
+    False otherwise. Objects created in the device memory are zero-initialised.
+
+    Parameters
+    ----------
+    obj : Indexed or Function or collection of Functions
+        The target object.
+    gpu_create : list of Function
+        The Function's which are expected to be created in device memory. This
+        information is given directly by the user through the compiler option
+        `devicecreate` and is propagated down here through the various stages of lowering.
+    """
+    try:
+        functions = (obj.function,)
+    except AttributeError:
+        functions = as_tuple(obj)
+
+    fcreate = []
+    for i in functions:
+        try:
+            f = i.alias or i
+        except AttributeError:
+            f = i
+        fcreate.append(f)
+
+    return all(f in gpu_create for f in fcreate)
```

Comment:
nitpicking in case list is "long". You don't actually need other `fcreate` you just need to return `false` at the first `i` not in `gpu_create` and return true at the end